ksqldb-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java

```diff
@@ -1092,6 +1106,48 @@ private LogicalSchema buildRepartitionedSchema(
     );
   }
 
+  /**
+   * Given a source node and its side, return {@code true} if the source should be repartitioned
+   * for the foreign key join.
+   *
+   * <p>The right hand side of the FK join must be repartitioned if it uses schema format baked by
+   * Schema Registry and the primary key is a struct. In that case, there is a possibility that
```

Comment:
> and the primary key is a struct

The PK should only be a struct for multi-column keys? Hence, such a a join should not be supported? Seems I am missing something? -- For what case we this actually happen?

Reply:
This can happen if we define the PK type as struct explicitly, see the added qtt test:

https://github.com/confluentinc/ksql/blob/aeae363c4c2d265241aa30583e121be01a81d5c0/ksqldb-functional-tests/src/test/resources/query-validation-tests/fk-join.json#L599-L604